Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -2261,43 +2271,38 @@ public AndConstraint<TAssertions> NotContainInOrder(params T[] unexpected)
         }
 
         IList<T> unexpectedItems = unexpected.ConvertOrCastToList();
-        IList<T> actualItems = Subject.ConvertOrCastToList();
-
-        if (unexpectedItems.Count > actualItems.Count)
+        if (unexpectedItems.Any())
         {
-            return new AndConstraint<TAssertions>((TAssertions)this);
-        }
-
-        var actualItemsSkipped = 0;
-        Func<T, T, bool> areSameOrEqual = ObjectExtensions.GetComparer<T>();
-        for (int index = 0; index < unexpectedItems.Count; index++)
-        {
-            T unexpectedItem = unexpectedItems[index];
-
-            actualItems = actualItems.SkipWhile(actualItem =>
-            {
-                actualItemsSkipped++;
-                return !areSameOrEqual(actualItem, unexpectedItem);
-            }).ToArray();
+            IList<T> actualItems = Subject.ConvertOrCastToList();
+            int subjectIndex = 0;
 
-            if (actualItems.Any())
+            Func<T, T, bool> areSameOrEqual = ObjectExtensions.GetComparer<T>();
```

Comment:
The bulk of this method and that of `NotContainInOrder` is the same. Considering the size of the body, I think it might be worthwhile to extract that into an extension method or private method. It'll make the algorithm clearer.